### DISCUSSION TITLE
Na meterwissel of verhuizing is jaarverbruik niet meer goed🚨

### DISCUSSION THREAD

**Discussion by rtrinks** [dsmrreader/dsmr-reader#1633] [Questions & answers / How-to's] [answered]
### Description

Er wordt aangegeven dat bij verhuizing of meterwissel, dat dit enkel de dag is waarop dat is uitgevoerd.
Maar op jaartotaal blijft dit een issue en heb ik een groot negatief getal wat dan logisch is.

Maar hoe lossen we dit op? Ik kan het nergens vinden en ook in de FAQ niet.

![Verbruik](https://user-images.githubusercontent.com/2185638/168465349-157316a8-f651-4cb8-8488-0982ef14fe5a.png)

### DSMR-reader version

DSMR-reader v4.18

**Comment by dennissiemensma** [comment]
Bedankt voor je vraag. Heb je de betreffende wisseldag zelf al gecorrigeerd?

Zie https://dsmr-reader.readthedocs.io/nl/v5/how-to/troubleshooting/smart-meter-replacement.html

Als het goed is klopt alles daarna weer, omdat het de maand- en jaartotalen een optelsom zouden moeten zijn van alle individuele dagen.

**Comment by rtrinks** [comment, marked as answer]
Aangepast en staat nu in ieder geval weer helemaal goed. Bedankt.

**Comment by claskfosmic** [comment]
Toch nog even reageren hier. Gebruik al een tijdje met veel plezier DSMR-reader. Van de week heeft Stedin de meter vervangen, dus nieuwe standen, dus het hele ding in de war. Om het e.e.a. te corrigeren heb ik handmatig alle reading van 00:00 t/m 11:05 (het moment dat oude meter uit ging) verwijderd. Vanaf c.a. 12:10 had ik weer nieuwe waardes. Ik had gehoopt dat daarmee alles weer zou kloppen, maar stond helaas nog steeds verkeerd.

Uiteindelijk via ReadTheDocs op Github uitgekomen en van daaruit hier terecht gekomen. Uiteindelijk toch even moeten tobben om zelf de dagstatistieken in te vullen. Ik had gelukkig de standen van de meter opgeschreven, dus op basis van de stand van 23:59:50 de dag ervoor heb ik het verbruik op de oude meter kunnen terughalen en met de nieuwe readings het verbruik op de nieuwe meter.

Na het invullen van de nieuwe standen, zag ik dat de kosten ook nog niet klopte. Ik snap dat dit een weinig voorkomende situatie is, maar het zou mooi zijn als DSMR-reader met het invullen van de nieuwe dagstatistieken, ook gelijk de nieuwe kosten + de totale kosten voor die dag kan uitrekenen.

Het is uiteindelijk gelukt, maar vind het nog behoorlijk omslachtig. Is DSMR-reader niet zelf in staat dit te detecteren? Ik neem toch aan dat het op basis van de berichten die uit de P1-poort komen, het mogelijk moet zijn om te detecteren dat het om een nieuwe meter gaat?

**Reply by dennissiemensma** [reply]
Bedankt voor je melding. De reden is dat het relatief veel werk is om dit sluitend te krijgen en zeker niet triviaal is. Helemaal wanneer er twee delen van dagen + een wisseling bij komt kijken. 

De architectuur van DSMR-reader is zo opgezet dat het vrij recht-toe-recht-aan is qua dataverwerking en berekeningen, met de consessie dat complexe of afwijkende situaties daardoor vaak niet kunnen of relatief veel moeite kosten. 
Deze situatie is daar een van en gezien de lage frequentie/kans per gebruiker dat het voor diegene voorkomt, ten opzichte van zelf voor 1 dag rekenen, blijft dit iets wat handwerk blijft.

Hierdoor is DSMR-reader relatief snel ontwikkeld, helemaal in de eerste jaren. Als ik de data opnieuw zou opzetten (achteraf), dan zou ik alles bijhouden als losse meters en zou een wisseling/vervanging minder impact hebben. Maar alle berekeningen zouden daardoor weer iets complexer zijn. Dus het is kiezen of delen.
